dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -0,0 +1,698 @@
+                    table_name=fq_table_name,
+                    schema=schema,
+                )
+
+    @lancedb_error
+    def is_storage_initialized(self) -> bool:
+        return self.table_exists(self.sentinel_table)
+
+    def _create_sentinel_table(self) -> Table:
+        """Create an empty table to indicate that the storage is initialized."""
+        return self.create_table(schema=NULL_SCHEMA, table_name=self.sentinel_table)
+
+    def _delete_sentinel_table(self) -> None:
+        """Delete the sentinel table."""
+        self.db_client.drop_table(self.sentinel_table)
+
+    @lancedb_error
+    def update_stored_schema(
+        self,
+        only_tables: Iterable[str] = None,
+        expected_update: TSchemaTables = None,
+    ) -> Optional[TSchemaTables]:
+        super().update_stored_schema(only_tables, expected_update)
+        applied_update: TSchemaTables = {}
+
+        try:
+            schema_info = self.get_stored_schema_by_hash(self.schema.stored_version_hash)
+        except DestinationUndefinedEntity:
+            schema_info = None
+
+        if schema_info is None:
+            logger.info(
+                f"Schema with hash {self.schema.stored_version_hash} "
+                "not found in the storage. upgrading"
+            )
+            self._execute_schema_update(only_tables)
+        else:
+            logger.info(
+                f"Schema with hash {self.schema.stored_version_hash} "
+                f"inserted at {schema_info.inserted_at} found "
+                "in storage, no upgrade required"
+            )
+        return applied_update
+
+    def get_storage_table(self, table_name: str) -> Tuple[bool, TTableSchemaColumns]:
+        table_schema: TTableSchemaColumns = {}
+
+        try:
+            fq_table_name = self.make_qualified_table_name(table_name)
+            arrow_schema = self.db_client.open_table(fq_table_name).schema
+        except FileNotFoundError:
+            return False, table_schema
+        except Exception:
+            raise
+
+        for field_name, field_type in arrow_schema_to_dict(arrow_schema).items():
+            schema_c: TColumnSchema = {
+                "name": self.schema.naming.normalize_identifier(field_name),
+                **self._from_db_type(field_type, None, None),
+            }
+            table_schema[field_type] = schema_c
+        return True, table_schema
+
+    @lancedb_error
+    def add_table_field(self, table_name: str, field_schema: TArrowField) -> Optional[Table]:
+        """Add a field to the LanceDB table.
+
+        Since arrow tables are immutable, this is done via a staging mechanism.
+        The data is stored in-memory in a staging arrow table, evolved then stored
+        written over the old table.
+
+        Args:
+            table_name: The name of the table to create the field on.
+            field_schema: The field to create.
+        """
+        arrow_table = self.db_client.open_table(table_name).to_arrow()
```

Comment:
does this mean all data needs to be loaded from the destination to create a field? we should definately add that to the docs page in a performance section or something like that.

Reply:
Will do.

I couldn't find a 'lancedb' way to do cloning. I'm sure there is a better way to do it. Shall we add an issue for this?